vxhunter/vxutility/symbol_table.py

```diff
@@ -197,3 +184,125 @@ def get_symtab_bounds(blk, vx_ver, symbol_fn):
     return symtab_start, symtab_end
 
 
+def find_sym_table(blk, start_addr, sym_st_fmt, sym_st_size, sym_types, vx_ver, verify_addresses=True):
+    i = start_addr
+    symtab = []
+    while i < blk.end.offset - sym_st_size:
+
+        sym, sym_valid = parse_sym(i, sym_st_fmt, sym_st_size, sym_types)
+
+        # if the current symbol struct is not a valid symbol, only increment the cursor by `word_size` bytes
+        # in case we are unaligned with the symbol table (this assumes the symbol table is word-aligned)
+        if sym is None or not sym_valid:
+            symtab = []
+            i += word_size
+            continue
+
+        name_addr = sym['name_addr']
+
+        # the name pointer must be within the bounds of the memory block
+        # if name_addr < blk.start.offset or name_addr >= blk.end.offset:
+        if verify_addresses:
```

Comment:
I don't know if it'd be too much work but it might be nice even if `verify_addresses=False` to make sure that the `max pointer - min pointer <= file size`